src/CMakeLists.txt

```diff
@@ -104,3 +104,7 @@ add_subdirectory(goto-instrument)
 add_subdirectory(goto-analyzer)
 add_subdirectory(goto-diff)
 add_subdirectory(goto-harness)
+
+if(WITH_MEMORY_ANALYZER)
+add_subdirectory(memory-analyzer)
+endif()
```

Comment:
Like the Makefile case, I'd prefer to link this to OS/build environment rather than introducing a new option.